File: Berger.Data/MsfaEntity/Users/UserInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Berger.Data.Attributes;
using Berger.Data.Common;

namespace Berger.Data.MsfaEntity.Users
{
    public class UserInfo : AuditableEntity<int>
    {
        public string FullName { get; set; }
        public string UserName { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public string Code { get; set; }
        public string EmployeeId { get; set; }
        public string Department { get; set; }
        public string Designation { get; set; }
        public string ManagerName { get; set; }
        public string ManagerId { get; set; }

        public string Address { get; set; }
        public string Gender { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public string ImageUrl { get; set; }

        public List<UserRoleMapping> Roles { get; set; }
        public List<UserZoneAreaMapping> UserZoneAreaMappings { get; set; }

        public UserInfo()
        {
            this.UserZoneAreaMappings = new List<UserZoneAreaMapping>();
        }
    }
}

File: BergerMsfaApi/ClientApp/src/app/Shared/Entity/Users/userInfo.ts
import { EnumEmployeeRole } from "../../Enums/employee-role";
import { Status } from "../../Enums/status";
import { QueryObject } from "../Common/query-object";

export class UserInfo {
    public id: number;
    public fullName: string;
    public userName: string;
    public email: string;
    public phoneNumber: string;
    // public code: string;
    public employeeId: string;
    public employeeRole: EnumEmployeeRole;
    public designation: string;
    public department: string;
    public managerName: string;
    public managerId: string;

    public gender: string;
    public address: string;
    public dateOfBirth?: Date;
    // public imageUrl: string;

    public status: number;
    public roleId: number;
    public roleName: string;
    public roleNames: string;

    public roleIds: number[];
    public plantIds: any[] = [];
    public areaIds: string[] = [];
    public saleOfficeIds: string[] = [];
    public territoryIds: string[] = [];
    public zoneIds: string[] = [];

    public statusText: string;
    public employeeRoleText: string;

    constructor(init?: Partial<UserInfo>) {
        this.roleIds = [];
        this.plantIds = [];
        this.zoneIds = [];
        this.areaIds = [];
        this.saleOfficeIds = [];
        this.plantIds = [];
        Object.assign(this, init);
    }

    clear() {
        this.id = null;
        this.status = Status.Active;
        this.employeeRole = null;
    }
}

export class SaveUserInfo {
    public id: number;
    public fullName: string;
    public userName: string;
    public email: string;
    public phoneNumber: string;
    // public code: string;
    public employeeId: string;
    public employeeRole: EnumEmployeeRole;
    public designation: string;
    public department: string;
    public managerName: string;
    public managerId: string;

    public gender: string;
    public address: string;
    public dateOfBirth?: Date;
    // public imageUrl: string;

    public status: number;
    public roleId: number;
    public roleName: string;

    public roleIds: number[];
    public plantIds: any[] = [];
    public areaIds: string[] = [];
    public saleOfficeIds: string[] = [];
    public territoryIds: string[] = [];
    public zoneIds: string[] = [];

    public statusText: string;

    constructor(init?: Partial<UserInfo>) {
        this.roleIds = [];
        this.plantIds = [];
        this.zoneIds = [];
        this.areaIds = [];
        this.saleOfficeIds = [];
        this.plantIds = [];
        Object.assign(this, init);
    }

    clear() {
        this.id = null;
        this.status = Status.Active;
        this.employeeRole = null;
    }
}

export class UserInfoQuery extends QueryObject {
    title: string;
    
    constructor(init?: Partial<UserInfoQuery>) {
        super();
        Object.assign(this, init);
    }

    clear() {
        this.title = '';
    }
}